presto-pinot-toolkit/src/test/java/com/facebook/presto/pinot/MetadataUtil.java

```diff
@@ -53,29 +49,25 @@ private MetadataUtil()
     public static final class TestingTypeDeserializer
             extends FromStringDeserializer<Type>
     {
-        private final Map<String, Type> types = ImmutableMap.of(
-                StandardTypes.BOOLEAN, BOOLEAN,
-                StandardTypes.BIGINT, BIGINT,
-                StandardTypes.INTEGER, INTEGER,
-                StandardTypes.DOUBLE, DOUBLE,
-                StandardTypes.VARCHAR, VARCHAR);
+        private final TypeManager typeManager;
```

Comment:
I couldn't figure out how this change to TypeManager is related. It is definitely a good change and avoids duplicating framework level information.

Reply:
This change is to json serialize/deserialize array type in the `PinotColumnHandle` 

The type gets serialized via `toString()` but while deserializing it we are only looking at primitive types here and using typemanager allows us to deserialize all types, i.e. array(varchar) , array(bigint) and so on.

Reply:
Got it. Its a testing only change so shouldn't matter much, but this would accept a wider set of types like Maps, nested arrays etc, more so than what the connector can currently take in. Thanks for explaining !

Reply:
Actually we may need this beyond testing, in the coordinator-worker model to pass column handles as json.

Reply:
Yeah good point: We should test this PR in the explicitly different coordinator-worker mode (ie regular mode) too. (The current MetadataUtil is test only apparently)

Also what about arrays and the PinotSegmentPageSource ?

Reply:
Added an implementation for `PinotSegmentPageSource` I believe the existing tests cover this change too, looking at rest of the comments.

Reply:
Hmm. I don't see any new tests in this PR about the segment page source multi-value support. How come the existing tests cover it ? IntelliJ"s run with code coverage is your friend :-P

Reply:
Actually all the new lines are covered out of the box in the coverage report as we have an all data types check here: https://github.com/prestodb/presto/blob/master/presto-pinot-toolkit/src/test/java/com/facebook/presto/pinot/TestPinotSegmentPageSource.java#L372
Although I don't really understand the hashcode check.

Reply:
Added an equality check for array values